core/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -222,6 +229,11 @@ private void raisePhaseFailure(SearchPhaseExecutionException exception) {
     @Override
     public final void onShardSuccess(Result result) {
         successfulOps.incrementAndGet();
+        long responseDuration = System.nanoTime() - startNanos;
+        String nodeId = result.getSearchShardTarget().getNodeId();
+        if (nodeId != null && collectorService != null) {
```

Comment:
why do we do this in here and not in the query wrapper. This action might run some DFS stats rather than queries so I think we should capture it solely in the `ResponseListenerWrapper`?